### PR TITLE
Add new types

### PR DESCRIPTION
**Pull request by bsudekum**
* Changes `.destination` -> `.text`
* Adds `.image`
* Changes how we parse out `.delimiter`

I'm thinking icon should become image.

/cc @mapbox/navigation-ios @lyzidiamond

### CODE REVIEW

File: MapboxDirections/MBVisualInstructionType.swift [hunk 1]
```diff
@@ -16,15 +16,22 @@ public enum VisualInstructionComponentType: Int, CustomStringConvertible {
     /**
      The component bears an exit number or the name of a place or street.
      */
-    case destination
+    case text
+    
+    /**
+     Component contains an icon that should be rendered.
+     */
+    case icon
```
Comment (1ec5) [inline MapboxDirections/MBVisualInstructionType.swift]: `image` tends to be used more often than `icon` in Cocoa frameworks. `image` would also highlight the fact that these components have non-nil `image` properties.

File: MapboxDirectionsTests/IntructionsTests.swift [hunk 2]
```diff
@@ -67,27 +68,24 @@ class SpokenInstructionsTests: XCTestCase {
         
         let spokenInstructions = step.instructionsSpokenAlongStep!
         
-        XCTAssertEqual(spokenInstructions[0].distanceAlongStep, 944.9)
-        XCTAssertEqual(spokenInstructions[1].distanceAlongStep, 258.0)
-        XCTAssertEqual(spokenInstructions[2].distanceAlongStep, 55.3)
-        
-        XCTAssertEqual(spokenInstructions[0].ssmlText, "<speak><amazon:effect name=\"drc\"><prosody rate=\"1.08\">Continue on Fillmore Street for a half mile</prosody></amazon:effect></speak>")
-        XCTAssertEqual(spokenInstructions[1].ssmlText, "<speak><amazon:effect name=\"drc\"><prosody rate=\"1.08\">In 900 feet, turn left onto <phoneme ph=\"ˈheɪt ˌstɹiːt\">Haight Street</phoneme></prosody></amazon:effect></speak>")
-        XCTAssertEqual(spokenInstructions[2].ssmlText, "<speak><amazon:effect name=\"drc\"><prosody rate=\"1.08\">Turn left onto <phoneme ph=\"ˈheɪt ˌstɹiːt\">Haight Street</phoneme></prosody></amazon:effect></speak>")
```
Comment (1ec5) [inline MapboxDirectionsTests/IntructionsTests.swift]: Consider adding assertions like these to ensure that we can still handle multiple spoken instructions on a single step.

Comment (bsudekum) [reply MapboxDirectionsTests/IntructionsTests.swift]: Added back https://github.com/mapbox/MapboxDirections.swift/pull/243/commits/babece99b5f76f561d1686a5a932f9c9797c873b

**Comment by bsudekum** [comment]
@1ec5 are you okay with changing `.destination -> .text`?

**Pull request merged**